src/content/en/updates/posts/2016/07/infinite-scroller.md

```diff
@@ -0,0 +1,246 @@
+      <rect class="whitener" x="800" y="-600" width="5000" height="1800"/>
+      <g class="viewport">
+        <rect x="0" y="0" width="800" height="600" />
+        <text x="-600" y="-16" transform="rotate(-90)" font-size="72">Viewport</text>
+      </g>
+      </g>
+    </svg>
+
+The same goes for scrolling in the other direction. We will, however, never shrink the runway in our implementation, so that the scrollbar position stays consistent.
+Tombstones
+As we mentioned earlier, we try to make our data source behave like something in the real world. With network latency and everything. That means that if our users make use of flicky scrolling, they can easily scroll past the last element we have data for. If that happens, we will place a tombstone item – a placeholder – that will get replaced by the item with actual content once the data has arrived. Tombstones are also recycled and have a separate pool for re-usable DOM elements. We need that so we can make a nice transition from a tombstone to the item populated with content, which would otherwise be very jarring to the user and might actually make them lose track of what they were focusing on.
+
+<img src="/web/updates/images/2016/07/infinite-scroller/tombstone.png" alt="Such tomb. Very stone. Wow.">
+
+An interesting challenge here is that real items can have a bigger height than the tombstone item because of differing amounts of text per item or an attached image. To resolve this, we will adjust the current scroll position every time data comes in and a tombstone is being replaced above the viewport, *anchoring* the scroll position to an element rather than a pixel value. This concept is called scroll anchoring.
+
+#Scroll Anchoring
+Our scroll anchoring will be invoked both when tombstones are being replaced as well as when the window gets resized (which also happens when the devices is being flipped!). We will have to figure out what the top-most visible element in the viewport is. As that element could only be partially visible, we will also store the offset from the top of the element where the viewport begins.
+
+<img src="/web/updates/images/2016/07/infinite-scroller/anchoring.png">
+
+After things in the runway have changed, we can now restore a situation that feels almost identical to the user. Win! Except a resized window means that each items has potentially changed its height, so how do we know how far down the anchored content should be placed? We don’t! To find out we would have to layout every element above the anchored item and add up all of their heights; this could cause a significant pause after a resize, and we don’t want that. Instead, we resort to assuming that every item above is the same size as a tombstone and adjust our scroll position accordingly. As elements are scrolled into the runway, we adjust or scroll position, effectively deferring the layout work to when it is actually needed.
```

Comment:
'As elements are scrolled into the runway' Do you mean as elements are scrolled into the viewport. 

'we adjust or scroll position' > 'we adjust our scroll position'.